lib/controllers/root-controller.js

```diff
@@ -574,6 +587,44 @@ export default class RootController extends React.Component {
     viewElement.focus();
   }
 
+  viewChangesForCurrentFile(stagingStatus) {
+    const editor = this.props.workspace.getActiveTextEditor();
+    const absFilePath = editor.getPath();
+    const repoPath = this.props.repository.getWorkingDirectoryPath();
+    if (absFilePath.startsWith(repoPath)) {
+      const filePath = absFilePath.slice(repoPath.length + 1);
+      this.quietlySelectItem(filePath, stagingStatus);
+      const splitDirection = this.props.config.get('github.viewChangesForCurrentFileDiffPaneSplitDirection');
+      const pane = atom.workspace.getActivePane();
+      if (splitDirection === 'right') {
+        pane.splitRight();
+      } else if (splitDirection === 'down') {
+        pane.splitDown();
+      }
+      return this.diveIntoFilePatchForPath(filePath, stagingStatus, {
+        lineNumber: editor.getCursorBufferPosition().row + 1,
+        amending: stagingStatus === 'staged' && this.state.isAmending,
+      });
+    } else {
+      throw new Error(`${absFilePath} does not belong to repo ${repoPath}`);
```

Comment:
Just adding my two cents — my gut reaction is to feel slightly uncomfortable with assuming that the active repository is the one we want to use for all operations. For example, I've wanted the ability to "pin" a repo so that it's always active for a while now. On a related note, I think that the diff views should render irrespective of the active repository; that is, they set and maintain their own context outside of the "active" one.